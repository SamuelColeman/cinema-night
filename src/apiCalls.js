export const currentMovies = async () => {
  const response = await fetch('https://api.themoviedb.org/3/movie/now_playing?api_key=2adea2e47475ecbf6312f332fc8e9ee2');
  const data = await response.json();
  data.results.forEach(movie => {
  	movie.poster_path = `https://image.tmdb.org/t/p/w500${movie.poster_path}`
    movie.backdrop_path = `https://image.tmdb.org/t/p/w500${movie.backdrop_path}`
  })
  return data.results;
}

export const loginVerification = async (info) => { 
  const options = {
    method: 'POST',
    body: JSON.stringify(info),
    headers: {
      'Content-Type': 'application/json'
    }
  }

  const resp = await fetch('http://localhost:3001/api/v1/login', options)
  const data = await resp.json();
  console.log('fetch data-->', data)
  return data
}

export const signUpVerification = async (info) => {
  const options = {
    method: 'POST',
    body: JSON.stringify(info),
    headers: {
      'Content-Type': 'application/json'
    }
  }

  const resp = await fetch('http://localhost:3001/api/v1/users', options)
  const data = await resp.json();
  console.log('fetch data-->', data)
  return data
}

export const addFavourite = async ( id, movie ) => {
  console.log('IN POST!!!', movie, id)
  const options = {
    method: 'POST',
    body: JSON.stringify(movie),
    headers: {
      'Content-Type': 'application/json'
    }
  }

  const resp =  await fetch(`http://localhost:3001/api/v1/users/${id}/moviefavorites`, options);
  const data = await resp.json();
  console.log(data)
  return data
}

export const getFavourites = async ( id ) => {

  const resp =  await fetch(`http://localhost:3001/api/v1/users/${id}/moviefavorites`);
  const data = await resp.json();
  return data
}

export const deleteFavorite = async ( userId, movieId ) => {
  const options = {
    method: 'DELETE',
    headers: {
      'Content-Type': 'application/json'
    }
  }

  const resp =  await fetch(`http://localhost:3001/api/v1/users/${userId}/moviefavorites/${movieId}`, options);
  const data = await resp.json();
  return data
}
