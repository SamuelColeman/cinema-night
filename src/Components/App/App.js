import React, { Component } from 'react';
import { currentMovies } from '../../apiCalls';
import { getMovies, isLoading, hasError } from '../../actions';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import MoviesContainer from '../MoviesContainer/MoviesContainer';
import Form from '../Form/Form';
import { Route } from 'react-router-dom';
import './App.css';

class App extends Component {

  async componentDidMount() {
    const { getMovies, hasError, isLoading } = this.props;
    try {
      isLoading(true);
      const movies = await currentMovies();
      isLoading(false)
      getMovies(movies);
    } catch (error) {
      hasError(error.message)
    }
  }

  

  render() {
    const { movies, errorMsg } = this.props;
    return (
      <section className='app'>
        {/* <MoviesContainer movies={movies} errorMsg={errorMsg} /> */}
        <Route exact path='/login' render={() => <Form /> } />
        <Form />
      </section>
    )
  }

}

export const mapStateToProps = ({movies, error}) => ({
  movies,
  error
});

export const mapDispatchToProps = (dispatch) => (
  bindActionCreators({
    getMovies,
    isLoading,
    hasError
  }, dispatch)
)



export default connect(mapStateToProps, mapDispatchToProps)(App);
