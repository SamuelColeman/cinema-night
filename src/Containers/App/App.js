import React, { Component } from 'react';
import { currentMovies, deleteFavorite, addFavourite, getFavourites } from '../../apiCalls';
import { getMovies, isLoading, hasError } from '../../actions';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types'
import MoviesContainer from '../MoviesContainer/MoviesContainer';
import MoviePage from '../MoviePage/MoviePage';
import FavouritesContainer from '../FavouritesContainer/FavouritesContainer';
import Form from '../Form/Form';
import { Route } from 'react-router-dom';
import './App.css'

class App extends Component {
  async componentDidMount() {
    const { getMovies, hasError, isLoading } = this.props;
    try {
      isLoading(true);
      const movies = await currentMovies();
      isLoading(false)
      getMovies(movies);
    } catch (error) {
      hasError(error.message)
    }
  }

  signOutUser = () => {
    let { currentUser, favouritesList } = this.props;
    currentUser.name = '';
    currentUser.id = null;
    currentUser.isSignedIn = false;
    favouritesList.favorites = [];
  }

  handleFavourite = (movie) => {
    let { currentUser, hasError } = this.props;
    if (currentUser.isSignedIn === true) {
      this.toggleFavourites(movie)
      hasError('');
    } else {
      hasError('Must be signed in to favourite!');
    }
  }

  toggleFavourites = async (movie) => {
    let { currentUser, favouritesList } = this.props;
    let currentMovie = favouritesList.favorites.find(film => film.title === movie.title);
    if (currentMovie === undefined) {
        document.getElementById(movie.title).setAttribute('class', 'active');
        this.displayFavourites(currentUser.id);
        addFavourite(movie, currentUser.id);
        this.displayFavourites(currentUser.id);
      } else {
        // document.getElementById(movie.title).removeAttribute('class');
        document.getElementById(movie.poster_path).setAttribute('hidden', 'true');
        this.removeFavourite(currentMovie);
      }
    this.displayFavourites(currentUser.id);
  }

  removeFavourite = async (movie) => {
    const { currentUser } = this.props; 
     try {
      await deleteFavorite(movie.user_id, movie.movie_id);
    } catch (error) {
      hasError(error.message)
    }
    this.displayFavourites(currentUser.id);
  }

  displayFavourites = async (id) => {
      let { favouritesList } = this.props;
      try {
        const resp = await getFavourites(id);
            favouritesList.favorites = resp.favorites;
          } catch (error) {
            console.log(error)
      }
  }

  render() {
    return (
      <section className='app'>
        <Route exact path='/login' render={() => <Form /> } />
        <Route exact path='/' render={() => <MoviesContainer handleFavourite={this.handleFavourite} signOutUser={this.signOutUser}/> } />
        <Route exact path='/favorites' render={() => <FavouritesContainer handleFavourite={this.handleFavourite}/>} />
        <Route exact path='/movies/:id' render={({ match }) => {
            let { movies } = this.props;
            const { id } = match.params;
            const filteredFilm = movies.find(movie => movie.id === parseInt(id)
            );
            return (
              <MoviePage {...filteredFilm} />
          )}} />
      </section>
    )
  }
}

export const mapStateToProps = (state) => ({
  movies: state.movies,
  error: state.error,
  currentUser: state.currentUser,
  favouritesList: state.favouritesList
});

export const mapDispatchToProps = (dispatch) => (
  bindActionCreators({
    getMovies,
    isLoading,
    hasError
  }, dispatch)
)

export default connect(mapStateToProps, mapDispatchToProps)(App);

App.propTypes = {
  movies: PropTypes.arrayOf(PropTypes.object).isRequired,
  currentUser: PropTypes.shape({
    name: PropTypes.string,
    id: PropTypes.number,
    isSignedIn: PropTypes.bool
  }),
  favorites: PropTypes.arrayOf(PropTypes.object),
  getMovies: PropTypes.func,
  hasError: PropTypes.func,
  isLoading: PropTypes.func
}

App.defaultProps = {
  isLoading: true,
  movies: [],
  currentUser: {},
  favorites: []
}