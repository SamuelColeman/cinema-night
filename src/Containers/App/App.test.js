import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import { currentMovies } from '../../apiCalls';

jest.mock('../../apiCalls');

describe('App', () => {

  // let wrapper;
  // let mockCurrentMovie;

  //   mockMovies = [{
  //     popularity: 462.91,
  //     vote_count: 3775,
  //     video: false,
  //     poster_path: "/udDclJoHjfjb8Ekgsd4FDteOkCU.jpg",
  //     id: 475557,
  //     adult: false,
  //     backdrop_path: "/n6bUvigpRFqSwmPp1m2YADdbRBc.jpg",
  //     original_language: "en",
  //     original_title: "Joker",
  //     genre_ids: [ 80, 18, 53 ],
  //     title: "Joker",
  //     vote_average: 8.6,
  //     overview: "During the 1980s, a failed stand-up comedian is driven insane and turns to a life of crime and chaos in Gotham City while becoming an infamous psychopathic crime figure.",
  //     release_date: "2019-10-04"
  //     },
  //     {
  //       popularity: 222.00,
  //       vote_count: 4000,
  //       video: false,
  //       poster_path: "/apath.jpg",
  //       id: 111111,
  //       adult: false,
  //       backdrop_path: "anotherpath.jpg",
  //       original_language: "en",
  //       original_title: "Mock Movie1",
  //       genre_ids: [ 80, 18, 53 ],
  //       title: "Mock Movie1",
  //       vote_average: 8.6,
  //       overview: "A description about a movie.",
  //       release_date: "2019-10-04"
  //       },
  //       {
  //         popularity: 100.00,
  //         vote_count: 2000,
  //         video: false,
  //         poster_path: "/udDclJoHjfjb8Ekgsd4FDteOkCU.jpg",
  //         id: 475557,
  //         adult: false,
  //         backdrop_path: "/n6bUvigpRFqSwmPp1m2YADdbRBc.jpg",
  //         original_language: "en",
  //         original_title: "Another movie",
  //         genre_ids: [ 80, 18, 53 ],
  //         title: "AnotherMovie",
  //         vote_average: 8.6,
  //         overview: "a really cool movie",
  //         release_date: "2019-10-04"
  //         }
  //   ]
  //   const mockDisplayFavorites = jest.fn();
  //   const mockUser = {
  //     name:'Pants',
  //     id:2,
  //     email:'pants@gmail.com',
  //   }

  //     currentMovies.mockImplementation(() => {
  //       return Promise.resolve(mockMovies);
  //     }); 
  //     addFavourite.mockImplementation(() => {
  //       return Promise.resolve(mockMovies)
  //     })
  //     removeFavourite.mockImplementation(() => {
  //       return Promise.resolve(mockMovies)
  //     })
  // })

  // beforeEach(() => {
  //   wrapper = shallow(<App 
  //       currentMovies = {mockSetMovies}
  //       addFavourite = {mockSetUpcomingMovies}
  //       removeFavourite = {mockSetFavorites}
  //       user = {mockUser}
  //       favorites= {[{
  //         "poster_path": "/tBuabjEqxzoUBHfbyNbd8ulgy5j.jpg",
  //         "movie_id": 420809,
  //         "title": "Maleficent: Mistress of Evil",
  //         "vote_average": 7.1,
  //         "overview": "Maleficent and her goddaughter Aurora begin to question the complex family ties that bind them as they are pulled in different directions by impending nuptials, unexpected allies, and dark new forces at play.",
  //         "release_date": "2019-10-18"
  //       }]}
  //       />)
  //     })

  // it.skip('should match snapshot with correct data passing throught', () => {
  //   expect(wrapper).toMatchSnapshot();
  // })

  // it.skip('should call getMovies when componentDidMount is called', () => {
  //   wrapper.instance().componentDidMount();

  //   expect(currentMovies).toHaveBeenCalled();
  // })

  // test isLoading to have been called
  // test hasError to have been called
  // test change of state, will need to go over redux testing class
})



  
