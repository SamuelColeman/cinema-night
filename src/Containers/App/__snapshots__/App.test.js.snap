// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot with correct data passing through 1`] = `
<section
  className="app"
>
  <Route
    exact={true}
    path="/login"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/favorites"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/movies/:id"
    render={[Function]}
  />
</section>
`;
